MaxText/checkpointing.py

```diff
@@ -16,51 +16,56 @@
 
 """Create an Orbax CheckpointManager with specified (Async or not) Checkpointer."""
 
+from typing import Optional
 from etils import epath
-import jax
-from orbax import checkpoint
-from orbax.checkpoint.checkpoint_manager import CheckpointManager, CheckpointManagerOptions, Checkpointer, AsyncCheckpointer
-from orbax.checkpoint import type_handlers
+from orbax.checkpoint.checkpoint_manager import CheckpointManager, CheckpointManagerOptions
+import orbax.checkpoint as ocp
+import grain.python as grain
 
 import max_logging
-
+from multihost_dataloading import MultiHostDataLoadIterator
 from flax.training import train_state
 
 def create_orbax_checkpoint_manager(
     checkpoint_dir: str,
     enable_checkpointing: bool,
     use_async: bool,
-    save_interval_steps: int
+    save_interval_steps: int,
+    dataset_type: Optional[str] = 'c4'
 ):
   """Returns specified Orbax (async or not) CheckpointManager or None if checkpointing is disabled."""
   if not enable_checkpointing:
     max_logging.log("Checkpointing disabled, not creating checkpoint manager.")
     return None
   max_logging.log("Creating checkpoint manager...")
   p = epath.Path(checkpoint_dir)
-  if use_async:
-    checkpointer = AsyncCheckpointer(checkpoint.PyTreeCheckpointHandler())
+
+  if dataset_type=='c4-array_record':
```

Comment:
Right now the user has two options: use TFRecord + TFDS input_pipeline, or ArrayRecord + grain input pipeline. Since dataset_type are tied to input_pipeline, I was trying to use one flag instead of two. And since we already have a flag for dataset_type, I just reuse it. So yes, currently setting  'dataset_type=c4-array_record' is the switch to turn on grain. It can be a bit confusing, but I was planning to document it in README and add note to base.yml later for clarification. Also open to other suggestions.